src/rt/lifetime.d

```diff
@@ -1096,15 +1118,19 @@ extern (C) void* _d_newitemU(in TypeInfo _ti)
     auto ti = unqualify(_ti);
     auto flags = !(ti.flags & 1) ? BlkAttr.NO_SCAN : 0;
     immutable tiSize = structTypeInfoSize(ti);
-    immutable size = ti.tsize + tiSize;
+    immutable itemSize = ti.tsize;
+    immutable size = itemSize + tiSize;
     if (tiSize)
         flags |= BlkAttr.STRUCTFINAL | BlkAttr.FINALIZE;
 
     auto blkInf = GC.qalloc(size, flags, ti);
     auto p = blkInf.base;
 
     if (tiSize)
+    {
+        *cast(TypeInfo*)(p + itemSize) = null; // the GC might not have cleared this area
```

Comment:
Is there some guarantee that there isn't anymore padding between this and the `ti` pointer at the end of the block?

Reply:
Everything else is already cleared by the GC:
https://github.com/dlang/druntime/blob/396a0ec7ef2ba38bb6314f7992d5d99be3645f02/src/gc/impl/conservative/gc.d#L417-L422

Reply:
This set is unaligned, breaking strict alignment targets.